policy/aggregation_type_config.go

```diff
@@ -36,41 +38,20 @@ type AggregationTypesConfiguration struct {
 	// Default aggregation types for gauge metrics.
 	DefaultGaugeAggregationTypes *AggregationTypes `yaml:"defaultGaugeAggregationTypes"`
 
-	// Metric suffix for aggregation type last.
-	LastSuffix *string `yaml:"lastSuffix"`
-
-	// Metric suffix for aggregation type sum.
-	SumSuffix *string `yaml:"sumSuffix"`
-
-	// Metric suffix for aggregation type sum square.
-	SumSqSuffix *string `yaml:"sumSqSuffix"`
-
-	// Metric suffix for aggregation type mean.
-	MeanSuffix *string `yaml:"meanSuffix"`
-
-	// Metric suffix for aggregation type min.
-	MinSuffix *string `yaml:"minSuffix"`
-
-	// Metric suffix for aggregation type max.
-	MaxSuffix *string `yaml:"maxSuffix"`
+	// Global type strings.
+	TypeStrings map[AggregationType]string `yaml:"typeStrings"`
 
-	// Metric suffix for aggregation type count.
-	CountSuffix *string `yaml:"countSuffix"`
+	// Type string overrides for Counter.
+	CounterTypeStringOverrides map[AggregationType]string `yaml:"counterTypeStringOverrides"`
 
-	// Metric suffix for aggregation type standard deviation.
-	StdevSuffix *string `yaml:"stdevSuffix"`
+	// Type string overrides for Timer.
+	TimerTypeStringOverrides map[AggregationType]string `yaml:"timerTypeStringOverrides"`
 
-	// Metric suffix for aggregation type median.
-	MedianSuffix *string `yaml:"medianSuffix"`
+	// Type string overrides for Gauge.
+	GaugeTypeStringOverrides map[AggregationType]string `yaml:"gaugeTypeStringOverrides"`
 
-	// Counter suffix overrides.
-	CounterSuffixOverrides map[AggregationType]string `yaml:"counterSuffixOverrides"`
-
-	// Timer suffix overrides.
-	TimerSuffixOverrides map[AggregationType]string `yaml:"timerSuffixOverrides"`
-
-	// Gauge suffix overrides.
-	GaugeSuffixOverrides map[AggregationType]string `yaml:"gaugeSuffixOverrides"`
+	// TypeStringTransformerType configs the type string transformer type.
+	TypeStringTransformerType string `yaml:"typeStringTransformerType"`
```

Comment:
`TransformType` is probably better? Update the comment as well.

Also instead of using a string, can we make it an enum type with custom unmarshal function and a `TransformFn` function that returns the corresponding transformation function? e.g.
```
type transformType int

const (
  unknownTransformType transformType
  identityTransform
  suffixTransform
)

func (t transformType) UnmarshalYAML(...) {...}

func (t transformType) TransformFn() (TransformFn, error)
```

Reply:
I'd prefer to keep it as string and do the parse way so that empty string(when user don't config this field) could be handled better, the UnmarshalYMAL way would require user to config this type even when they only need the default noop transformFn

Reply:
Hm can't you make it a *transformType in the config? If it's nil, don't set it.